jscomp/others/node_buffer.ml

```diff
@@ -26,17 +26,208 @@
+external readUInt32BE : t -> int -> int = "readUInt32BE" [@@bs.send]
+external readUInt32LE : t -> int -> int = "readUInt32LE" [@@bs.send]
+external readUIntBE : t -> offset:int -> byteLength:int -> int = "readUIntBE" [@@bs.send]
+external readUIntLE : t -> offset:int -> byteLength:int -> int = "readUIntLE" [@@bs.send]
+
+external subarray : t -> ?start: int -> ?end_: int -> unit -> t = "subarray" [@@bs.send]
+external slice : t -> ?start: int -> ?end_:int -> unit -> t = "slice" [@@bs.send]
+
+external swap16 : t -> t = "swap16" [@@bs.send]
+external swap32 : t -> t = "swap32" [@@bs.send] 
+external swap64 : t -> t = "swap64" [@@bs.send] 
+
+external toJSON : t -> Js.Json.t = "toJSON" [@@bs.send]
+
+external write :
+  t ->
+  string ->
+  ?offset:int ->
+  ?length:int ->
+  ?encoding: ([ `ascii  | `utf8  | `utf16le  | `usc2  | `base64  | `latin1 | `binary | `hex ] [@bs.string]) ->
+  unit -> int = "write" [@@bs.send]
+
+external writeDoubleBE : t -> float -> offset:int -> int = "writeDoubleBE" [@@bs.send]
+external writeDoubleLE : t -> float -> offset:int -> int = "writeDoubleLE" [@@bs.send]
+external writeFloatBE : t -> float -> offset:int -> int = "writeFloatBE" [@@bs.send]
+external writeFloatLE : t -> float -> offset:int -> int = "writeFloatLE" [@@bs.send]
+
+external writeInt8 : t -> int -> offset:int -> int = "writeInt8" [@@bs.send]
+external writeInt16BE : t -> int -> offset:int -> int = "writeInt16BE" [@@bs.send]
+external writeInt16LE : t -> int -> offset:int -> int = "writeInt16LE" [@@bs.send]
+external writeInt32BE : t -> int -> offset:int -> int = "writeInt32BE" [@@bs.send]
+external writeInt32LE : t -> int -> offset:int -> int = "writeInt32LE" [@@bs.send]
+external writeIntBE : t -> int -> offset:int -> byteLength:int -> int = "writeIntBE" [@@bs.send]
+external writeIntLE : t -> int -> offset:int -> byteLength:int -> int = "writeIntLE" [@@bs.send]
+
+external writeUInt8 : t -> int -> offset:int -> int = "writeUInt8" [@@bs.send]
+external writeUInt16BE : t -> int -> offset:int -> int = "writeUInt16BE" [@@bs.send]
+external writeUInt16LE : t -> int -> offset:int -> int = "writeUInt16LE" [@@bs.send]
+external writeUInt32BE : t -> int -> offset:int -> int = "writeUInt32BE" [@@bs.send]
+external writeUInt32LE : t -> int -> offset:int -> int = "writeUInt32LE" [@@bs.send]
+external writeUIntBE : t -> int -> offset:int -> byteLength:int -> int = "writeUIntBE" [@@bs.send]
+external writeUIntLE : t -> int -> offset:int -> byteLength:int -> int = "writeUIntLE" [@@bs.send]
+
+external _INSPECT_MAX_BYTES : t -> int = "INSPECT_MAX_BYTES" [@@bs.get]
+external kMaxLength : t -> int = "kMaxLength" [@@bs.get]
+
+external transcode : 
+  t ->
+  fromEnc: ([ `ascii  | `utf8  | `utf16le  | `usc2  | `base64  | `latin1 | `binary | `hex ] [@bs.string]) ->
+  toEnc: ([ `ascii  | `utf8  | `utf16le  | `usc2  | `base64  | `latin1 | `binary | `hex ] [@bs.string]) ->
+  t = "transcode"
+[@@bs.val] [@@bs.scope "Buffer"]
```

Comment:
For some weird reason, `transcode` is apparently not a class method of the `Buffer` class but actually a property of the `buffer` module: https://nodejs.org/api/buffer.html#buffer_buffer_transcode_source_fromenc_toenc

So instead of using `[@@bs.val] [@@bs.scope "Buffer"]` you can use `[@@bs.module "buffer"]`. Same for `INSPECT_MAX_BYTES`, `kMaxLength`, and everything in `constants`!

Reply:
Great, will fix, thank you!

Reply:
fixed